pyvista/core/utilities/geometric_objects.py

```diff
@@ -383,6 +396,524 @@ def Sphere(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
+
+    """
+    if end_theta is None:
+        end_theta = 2 * np.pi if radians else 360.0
+    if end_phi is None:
+        end_phi = np.pi if radians else 180.0
+
+    radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    theta = np.linspace(start_theta, end_theta, theta_resolution)
+    phi = np.linspace(start_phi, end_phi, phi_resolution)
+    return SolidSphereGeneric(
+        radius,
+        theta,
+        phi,
+        center,
+        direction,
+        radians=radians,
+        tol_radius=tol_radius,
+        tol_angle=tol_angle,
+    )
+
+
+def SolidSphereGeneric(
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+    radians=False,
+    tol_radius=1.0e-8,
+    tol_angle=None,
+):
+    """Create a solid sphere with flexible sampling.
+
+    A solid sphere fills space in 3D in comparison to
+    :func:`pyvista.Sphere`, which is a 2D surface.
+
+    This function allows user defined sampling of each spherical
+    coordinate, whereas :func:`pyvista.SolidSphere`
+    only allows linear sampling.   Angles are by default
+    specified in degrees.
+
+    PyVista uses a convention where ``theta`` represents the azimuthal
+    angle (similar to degrees longitude on the globe) and ``phi``
+    represents the polar angle (similar to degrees latitude on the
+    globe). In contrast to latitude on the globe, here
+    ``phi`` is 0 degrees at the North Pole and 180 degrees at the South
+    Pole. ``phi=0`` is on the positive z-axis by default.
+    ``theta=0`` is on the positive x-axis by default.
+
+    Parameters
+    ----------
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. Must have at least two points and be non-negative.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``theta``
+        points. Must have at least two points.  Can have any value as long
+        as range is within 360 degrees. Large magnitudes may result in
+        problems with endpoint overlap detection.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``phi``
+        points. Must have at least two points.  Must be between
+        0 and 180 degrees.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center coordinate vector in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction coordinate vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees ``phi``.
+
+    radians : bool, default: False
+        Whether to use radians for ``theta`` and ``phi``. Default is degrees.
+
+    tol_radius : bool, default: 1.0e-8
```

Comment:
```suggestion
    tol_radius : float, default: 1.0e-8
```
perhaps with just `optional`.